src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.cs

```diff
@@ -320,6 +321,250 @@ public async Task binding_in_variable_value_is_valid()
+            Content-Type: application/xml
+            
+            {
+                "review_count" : "{{$randomInt 10 99}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+        var bodyAsString = await request.Content.ReadAsStringAsync();
+        var randIntValue = bodyAsString.Split(":").Last().Trim().Substring(1, 2);
+
+
+        int intValueOfRandInt = int.Parse(randIntValue);
+
+        intValueOfRandInt.Should().BeGreaterThanOrEqualTo(10);
+        intValueOfRandInt.Should().BeLessThanOrEqualTo(99);
+    }
+
+    [Fact]
+    public async Task can_bind_datetime_with_parameters()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpKernel(client: client);
+
+        using var _ = new AssertionScope();
+
+        var code = """
+            POST https://api.example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            
+            {
+                "custom_date" : "{{$datetime 'yyyy-MM-dd'}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+        var currentDate = DateTime.UtcNow.ToString("yyyy-MM-dd");
+        var bodyAsString = await request.Content.ReadAsStringAsync();
+        bodyAsString.Should().BeExceptingWhitespace($$"""
+            {
+                "custom_date" : "{{currentDate}}"
+            }
+            """);
+
+    }
+
+    [Fact]
+    public async Task can_bind_local_datetime_with_parameters()
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpKernel(client: client);
+
+        using var _ = new AssertionScope();
+
+        var code = """
+            POST https://api.example.com/comments HTTP/1.1
+            Content-Type: application/xml
+            
+            {
+                "local_custom_date" : "{{$localDatetime 'yyyy-MM-dd'}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+        var currentDate = DateTime.UtcNow.ToString("yyyy-MM-dd");
+        var bodyAsString = await request.Content.ReadAsStringAsync();
+        bodyAsString.Should().BeExceptingWhitespace($$"""
```

Comment:
Consider: Similar comment about splitting and comparing just the date part so that you can directly compare with the expected date as opposed to creating a string with the expected date embedded and compating the string.